drivers/VirtualWatchdog.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019 Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_VIRTUAL_WATCHDOG_H
+#define MBED_VIRTUAL_WATCHDOG_H
+
+#ifdef DEVICE_WATCHDOG
+
+#include <cstdio>
+#include "platform/mbed_error.h"
+#include "platform/mbed_critical.h"
+#include "platform/mbed_power_mgmt.h"
+#include "platform/mbed_assert.h"
+#include "platform/SingletonPtr.h"
+
+#if DEVICE_LPTICKER
+#include "drivers/LowPowerTicker.h"
+#else
+#include "drivers/Ticker.h"
+#endif
+namespace mbed {
+
+/** \addtogroup drivers */
+
+/** VirtualWatchdog should be used for applications where you use multiple services requiring watchdog functionality.
```

Comment:
We should probably mention that using both Watchdog and VirtualWatchdog at the same time is not supported.

Reply:
will add